neuralpp/test/quick_tests/graphical_model/representation/factor/symbolic_factor_test.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+
+import sympy
+from typing import List, Any, Optional, Type, Callable, Dict
+from neuralpp.inference.graphical_model.representation.factor.symbolic_factor import SymbolicFactor
+from neuralpp.symbolic.sympy_expression import SymPyVariable, SymPyFunctionApplication
+from neuralpp.inference.graphical_model.variable.integer_variable import IntegerVariable
+
+int_to_int_to_int = Callable[[int, int], int]
+
+def test_sympy_condition():
+    x = IntegerVariable("x", 3)
+    y = IntegerVariable("y", 2)
+
+    x_sympy, y_sympy = sympy.symbols("x y")
+    expression1 = SymPyFunctionApplication(x_sympy * y_sympy, {x_sympy: int, y_sympy: int})
```

Comment:
> You should be able to use `expression1 = x_sympy * y_sympy` here since there is operator overloading.
I don't believe I can, since ` x_sympy` and `y_sympy` are both sympy variables, and not `expressions`. I could make them as expressions and then use operator overloading if that's better.